### ISSUE TITLE
Segfault with missing alias in duckdb_scanner

### ISSUE THREAD

**Issue by prrao87** [kuzudb/kuzu#3318]
The duckdb scanner segfaults if there's a missing alias. Example to reproduce is below. 👇🏽

This segfaults:

```
kuzu> LOAD EXTENSION duckdb_scanner;
kuzu> ATTACH '/Users/prrao/code/kuzu/build/release/tools/shell/test.db' (dbtype 'duckdb'); 
[1]    8153 segmentation fault  ./kuzu_shell test_db
```

What's missing is the `AS testdb` alias when we attach the DB. This works:

```
kuzu> ATTACH '/Users/prrao/code/kuzu/build/release/tools/shell/test.db' AS testdb (dbtype 'duckdb');
---------------------------------
| result                        |
---------------------------------
| Attach database successfully. |
---------------------------------
(1 tuple)
(1 column)
Time: 0.06ms (compiling), 0.24ms (executing)
```

We should gracefully handle the missing alias error.

**Comment by prrao87** [comment]
@acquamarin I tried running this sample DB with the following data:

```
┌───────┬─────┐
│ name  ┆ age │
│ ---   ┆ --- │
│ str   ┆ i32 │
╞═══════╪═════╡
│ Wendy ┆ 28  │
│ Jeff  ┆ 32  │
└───────┴─────┘
```
DuckDB database file:

[ducktest.db.zip](https://github.com/kuzudb/kuzu/files/15030317/ducktest.db.zip)

Going through this workflow, I get a segfault despite the correct syntax this time for the attach statement. Could you try reproducing on the attached DB and see if it works?

```
kuzu> INSTALL duckdb_scanner;
kuzu> LOAD EXTENSION duckdb_scanner;
-
(0 tuples)
(0 columns)
Time: 0.01ms (compiling), 8.92ms (executing)
kuzu> ATTACH '/Users/prrao/code/kuzu/build/release/tools/shell/ducktest.db' AS mydb (dbtype 'duckdb');
[1]    16911 segmentation fault  ./kuzu_shell test_db
```

**Comment by acquamarin** [comment]
```
➜  shell git:(fix-load-duckdb) ✗ ./kuzu_shell dsadad
Opened the database at path: dsadad in read-write mode.
Enter ":help" for usage hints.
kuzu> install duckdb_scanner;
-
(0 tuples)
(0 columns)
Time: 0.08ms (compiling), 1954.45ms (executing)
kuzu> load extension duckdb_scanner;
-
(0 tuples)
(0 columns)
Time: 0.03ms (compiling), 58.61ms (executing)
kuzu> attach '/tmp/1.db' (dbtype 'duckdb');
---------------------------------
| result                        |
---------------------------------
| Attach database successfully. |
---------------------------------
(1 tuple)
(1 column)
Time: 0.09ms (compiling), 88.84ms (executing)
```
Should work now.
Can you double check whether it works on your side?

**Comment by prrao87** [comment]
What duckdb database did you use for the above example? The one I shared? I think the issue is when I create the duckdb database from Python.

**Comment by prrao87** [comment]
Fixed by #3324.
